test/autorest/additionalpropsgroup/zz_generated_models.go

```diff
@@ -96,14 +104,20 @@ type PetAPInPropertiesResponse struct {
 }
 
 type PetAPInPropertiesWithAPString struct {
-	// Contains additional key/value pairs not defined in the schema.
+	// ID - REQUIRED
+	ID *int32 `json:"id,omitempty"`
+
+	// OdataLocation - REQUIRED
+	OdataLocation *string `json:"@odata.location,omitempty"`
+
+	// OPTIONAL; Contains additional key/value pairs not defined in the schema.
 	AdditionalProperties map[string]*string
 
-	// Dictionary of
+	// OPTIONAL; Dictionary of
 	AdditionalProperties1 map[string]*float32 `json:"additionalProperties,omitempty"`
-	ID                    *int32              `json:"id,omitempty"`
-	Name                  *string             `json:"name,omitempty"`
-	OdataLocation         *string             `json:"@odata.location,omitempty"`
+
+	// Name - OPTIONAL
```

Comment:
What's the difference between some having the OPTIONAL/REQUIRED/READ-ONLY at the beginning vs some after the name?  Should this be more homogenous? It looks a little weird that the Optional and required switch places in some cases.

Reply:
Also, unrelated to this PR but the `Dictionary of` is missing something?

Reply:
The `Dictionary of` looks to be an issue with M4.  We don't generate this, nor is it in the swagger.  I'll follow up on that.

The modifier showing up as a suffix is for cases where the property has no doc comment.  We did it like this in track 1 as simply having `// READ-ONLY` would make the linter unhappy.  That no longer appears to be the case.  For consistency, I can remove this.

Reply:
hmm, yeah at least for now, if the linter doesnt complain about it, then I vote for more consistency in the comments.

Reply:
Also I think I misspoke about the `Dictionary of` thing, it looks like we're somehow munging the description.  I'll fix that too.

Reply:
Fixing the `Dictionary of ` looks to be more involved than I thought.  I've opened https://github.com/Azure/autorest.go/issues/623 to track it.

Reply:
The suffix thing has been fixed.